include/mbgl/style/light.hpp

```diff
@@ -0,0 +1,294 @@
+#pragma once
+
+#include <mbgl/style/types.hpp>
+#include <mbgl/style/property_value.hpp>
+#include <mbgl/style/property_evaluator.hpp>
+#include <mbgl/style/property_evaluation_parameters.hpp>
+#include <mbgl/style/transition_options.hpp>
+#include <mbgl/style/cascade_parameters.hpp>
+#include <mbgl/util/constants.hpp>
+#include <mbgl/util/color.hpp>
+#include <mbgl/util/constants.hpp>
+#include <mbgl/util/interpolate.hpp>
+#include <mbgl/util/indexed_tuple.hpp>
+#include <mbgl/util/ignore.hpp>
+
+#include <memory>
+
+namespace mbgl {
+namespace style {
+
+template <class Value>
+class UnevaluatedLightProperty {
```

Comment:
Rather than introduce a new class, let's generalize the existing `UnevaluatedPaintProperty` so that it can be used here. The functionality is solid, it just needs to be extracted from `paint_property.hpp` into its own header, and given a name that does not tie it to paint properties specifically. I suggest `TransitioningProperty`, since its main responsibility is to interpolate between a current and prior value based on the current clock and transition options.